app/models/hackney/repair_request.rb

```diff
@@ -27,4 +22,44 @@ def telephone_number
   def contact_name
     contact&.name
   end
+
+  #
+  # persistence
+  #
+  def work_orders_attributes=(a)
+    a = a.values if a.is_a?(Hash)
+    self.work_orders = a.map {|x| Hackney::WorkOrder.new(x)}
+  end
+
+  def contact_attributes=(a)
+    self.contact = Hackney::Contact.new(a)
+  end
+
+  def save
+    response = HackneyAPI::RepairsClient.new.post_repair_request(
+      name: contact_name,
+      phone: telephone_number,
+      sor_code: work_orders&.first&.sor_code,
+      priority: priority,
+      property_ref: property_reference,
+      description: description
+    )
+
+    response.present? or raise "*** API ERROR SAVING REPAIR REQUEST. response: #{response} ***" # TODO
+    self.attributes_from_api = response
+  end
+
+  #
+  # read attributes from API
+  #
+  def attributes_from_api=(a)
+    self.attributes = {
```

Comment:
I think I would prefer this to be a class method and return the attributes rather than mutate the state. This way mutating state is explicit in `#save` and `.build`. I'm not sure why I prefer it that way though.

Reply:
What do you mean by "return the attributes"? This method is specifically for mutating the state, like `#attribute=`, but accepting the api format.

I just extracted it from `#build`.

Reply:
Ohhh maybe you mean making a class method that just maps the api format into a hash of attributes?

Reply:
yup

Reply:
ok I'll do that then!

Reply:
done!